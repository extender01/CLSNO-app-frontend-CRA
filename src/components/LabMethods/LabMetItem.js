import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

const LabMetItem = (props) => (
    <div  className={`lm__item lm__${props.category}`}>
        
       
        {/* if filtering text exists in redux, check if that searched text string is contained in name of labTest, if so then apply class to div*/}
        <div className={(props.textFilter && props.name.toLowerCase().includes(props.textFilter.toLowerCase())) ? 'lm__item--found' : undefined}>

            {props.category === 'external' && (
                <Link to={{
                    pathname: '/metody/detail/' + props._id,
                    state: 'external' 
                }}>
                    {props.name}
                </Link>

            )}

            {props.category === 'internal' && (
                <Link to={{
                    pathname: '/metody/detail/' + props._id,
                    state: 'internal' 
                }}>
                    {props.name}
                </Link>

            )}

            {props.category === 'calculated' && (
                <Link to={{
                    pathname: '/metody/detail/' + props._id,
                    state: 'calculated' 
                }}>
                    {props.name}
                </Link>

            )}


        </div>

        <div className={(props.textFilter && props.syn.toLowerCase().includes(props.textFilter.toLowerCase())) ? 'lm__item--found' : undefined}>
            <p>{props.syn}</p>
        </div>

        <div>
            <Link to={'/edit/' + props._id}>{props.where}</Link>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        textFilter: state.labTests.filters.text
    };
};

LabMetItem.propTypes = {
    category: PropTypes.string,
    textFilter: PropTypes.string,
    name: PropTypes.string,
    syn: PropTypes.string,
    where: PropTypes.string,
    _id: PropTypes.string
};


export default connect(mapStateToProps)(LabMetItem);

