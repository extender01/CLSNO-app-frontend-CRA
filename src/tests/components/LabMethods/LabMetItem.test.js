import React from 'react';
import {shallow} from 'enzyme';
import LabMetItem from '../../../components/LabMethods/LabMetItem';
import {labmets} from '../../fixtures/labmets';


test('should render Lab Met Item correctly', () => {
    const wrapper = shallow(<LabMetItem {...labmets[1]} />);
    expect(wrapper).toMatchSnapshot();
});