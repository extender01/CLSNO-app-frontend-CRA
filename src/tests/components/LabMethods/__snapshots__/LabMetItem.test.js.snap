// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Lab Met Item correctly 1`] = `
<div
  className="f_between lm__item"
>
  <div
    className=""
  >
    <Link
      replace={false}
      to={
        Object {
          "pathname": "/methods/detail/5bea74c7c75591445c168693",
          "state": "external",
        }
      }
    >
      CYFRA
    </Link>
  </div>
  <div>
    <p>
      cyp je to
    </p>
  </div>
  <div>
    <Link
      replace={false}
      to="/edit/5bea74c7c75591445c168693"
    >
      Krnov
    </Link>
  </div>
</div>
`;
